Assets/Scripts/HealthBar.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class HealthBar : MonoBehaviour
+{
+    public GameObject Entity;
+    private RectTransform Transform;
+    // Start is called before the first frame update
+    void Start()
+    {
+        Transform = GetComponent<RectTransform>();
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        SizeHealthBar(); // keep checking the size
+    }
+
+    public void SizeHealthBar() // function to check the healthbar and resize it if necesary
+    {
+        int currentHealth = Entity.GetComponent<Entity>().Health; // get currenthealth of entitiy
```

Comment:
zou je deze components niet beter 1 x op kunnen halen i.p.v. telkens getcomponent te doen?

Reply:
maakt  niet veel verschil, maar zal dat wel effe doen

Reply:
aangepast, kan je checken of dit beter is